dwave/cloud/client.py

```diff
@@ -901,7 +901,12 @@ def _handle_problem_status(self, message, future):
                 future._set_error(CanceledFutureError())
             else:
                 # Return an error to the future object
-                future._set_error(SolverFailureError(message.get('error_message', 'An unknown error has occurred.')))
+                errmsg = message.get('error_message', 'An unknown error has occurred.')
+                if 'solver is offline' in errmsg.lower():
+                    err = SolverOfflineError(errmsg)
+                else:
+                    err = SolverFailureError(errmsg)
```

Comment:
How come you're not checking for the other known errors, like authentication?

Reply:
They are checked for before.